services/src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -104,9 +106,10 @@ public ProduceService(Map<String, Object> props, String name) throws Exception {
 
     if (existingPartitionCount <= 0) {
       if (config.getBoolean(ProduceServiceConfig.TOPIC_CREATION_ENABLED_CONFIG)) {
+        TopicFactory topicFactory = config.getConfiguredInstance(ProduceServiceConfig.TOPIC_FACTORY_CONFIG, TopicFactory.class);
         _partitionNum.set(
-            Utils.createMonitoringTopicIfNotExists(_zkConnect, _topic, topicReplicationFactor,
-                partitionsToBrokersRatio));
+            topicFactory.createTopic(_zkConnect, _topic, topicReplicationFactor,
+                partitionsToBrokersRatio, props));
```

Comment:
On master I thought the topic creation was still in the ProduceService and so the parameters will be there.  Warren has not yet completed the PR to move the topic creation into the TopicManagementService .  When that is done properties should always be non-null and populated.